operators/src/adapters/raster_time.rs

```diff
@@ -0,0 +1,457 @@
+    St1: Stream<Item = Result<RasterTile2D<T1>>>,
+    St2: Stream<Item = Result<RasterTile2D<T2>>>,
+    F1: Fn(QueryRectangle) -> St1,
+    F2: Fn(QueryRectangle) -> St2,
+{
+    pub fn new(source_a: F1, source_b: F2, query_rect: QueryRectangle) -> Self {
+        Self {
+            stream: source_a(query_rect).zip(source_b(query_rect)),
+            source_a,
+            source_b,
+            query_rect,
+            time_end: None,
+        }
+    }
+
+    // TODO: move to TimeInterval
+    fn intersect_time_intervals(t1: TimeInterval, t2: TimeInterval) -> Option<TimeInterval> {
+        if t1.intersects(&t2) {
+            let start = std::cmp::max(t1.start, t2.start);
+            let end = std::cmp::min(t1.end, t2.end);
+            Some(TimeInterval::new_unchecked(start, end))
+        } else {
+            None
+        }
+    }
+
+    fn align_tiles(
+        mut tile_a: RasterTile2D<T1>,
+        mut tile_b: RasterTile2D<T2>,
+    ) -> (RasterTile2D<T1>, RasterTile2D<T2>) {
+        // TODO: handle error
+        let time = Self::intersect_time_intervals(tile_a.time, tile_b.time).unwrap();
+        tile_a.time = time;
+        tile_b.time = time;
+        (tile_a, tile_b)
+    }
+}
+
+impl<T1, T2, St1, St2, F1, F2> Stream for RasterTimeAdapter<T1, T2, St1, St2, F1, F2>
+where
+    T1: Pixel,
+    T2: Pixel,
+    St1: Stream<Item = Result<RasterTile2D<T1>>>,
+    St2: Stream<Item = Result<RasterTile2D<T2>>>,
+    F1: Fn(QueryRectangle) -> St1,
+    F2: Fn(QueryRectangle) -> St2,
+{
+    type Item = Result<(RasterTile2D<T1>, RasterTile2D<T2>)>;
+
+    #[allow(clippy::needless_return)]
+    fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        let RasterTimeAdapterProjection {
+            source_a,
+            source_b,
+            query_rect,
+            time_end,
+            mut stream,
+        } = self.project();
+
+        // TODO: handle error
+        let next = ready!(stream.as_mut().poll_next(cx)).map(|(a, b)| (a.unwrap(), b.unwrap()));
+
+        if let Some((tile_a, tile_b)) = next {
+            if let Some(end) = time_end {
+                if tile_a.time.start >= end.0 || tile_b.time.start >= end.1 {
+                    // a tile belongs to next slice => query next time slice of sources instead
+                    // TODO: determine end of time slice in stream without computing an additional tile
+                    let mut next_qrect = *query_rect;
+                    next_qrect.time_interval.start = min(end.0, end.1);
+                    *time_end = None;
+
+                    stream.set(source_a(next_qrect).zip(source_b(next_qrect)));
+
+                    // TODO: handle error
+                    let next = ready!(stream.as_mut().poll_next(cx))
+                        .map(|(a, b)| (a.unwrap(), b.unwrap()));
+                    if let Some((tile_a, tile_b)) = next {
+                        return Poll::Ready(Some(Ok(Self::align_tiles(tile_a, tile_b))));
```

Comment:
add a todo to scale data if required.

Reply:
f7ac443a1b83523535362901d8f86e08bba8de8b